src/main/java/fr/xephi/authme/listener/PlayerListener.java

```diff
@@ -190,46 +192,73 @@ public void onPlayerJoin(PlayerJoinEvent event) {
         management.performJoin(player);
     }
 
+    private void runOnJoinChecks(String name, String ip) throws FailedVerificationException {
+        // Fast stuff
+        onJoinVerifier.checkSingleSession(name);
+        onJoinVerifier.checkIsValidName(name);
+
+        // Get the auth later as this may cause the single session check to fail
+        // Slow stuff
+        final PlayerAuth auth = dataSource.getAuth(name);
+        final boolean isAuthAvailable = auth != null;
+        onJoinVerifier.checkKickNonRegistered(isAuthAvailable);
+        onJoinVerifier.checkAntibot(name, isAuthAvailable);
+        onJoinVerifier.checkNameCasing(name, auth);
+        onJoinVerifier.checkPlayerCountry(name, ip, isAuthAvailable);
```

Comment:
One iteration beyond this (but only once we resolve this permission behavior change!) is to look at whether we really need the auth object at all? As far as I see here from a quick look, we only want to know whether `isAuthAvailable` and then I only see it used in `checkNameCasing(name, auth)`.
If instead we can have a dedicated method to return the real name (similar to `DataSourceResult<String> getEmail(String)`) we avoid constructing PlayerAuth objects, which is more costly.